src/overlays/actors/ovl_En_Ma_Yts/z_en_ma_yts.c

```diff
@@ -21,48 +74,519 @@ const ActorInit En_Ma_Yts_InitVars = {
     (ActorFunc)EnMaYts_Update,
     (ActorFunc)EnMaYts_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma_Yts_0x80B8D030/func_80B8D030.asm")
+void EnMaYts_UpdateEyes(EnMaYts* this) {
+    if (this->overrideEyeTexIndex != 0) {
+        this->eyeTexIndex = this->overrideEyeTexIndex;
+    } else if (DECR(this->blinkTimer) == 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex >= 3) {
+            this->blinkTimer = Rand_S16Offset(30, 30);
+            this->eyeTexIndex = 0;
+        }
+    }
+}
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+    { 18, 46, 0, { 0, 0, 0 } },
+};
+
+static CollisionCheckInfoInit2 D_80B8E19C = {
+    0, 0, 0, 0, 0xFF,
+};
+
+struct struct_80B8E1A8 {
+    /* 0x00 */ AnimationHeader* unk_00;
+    /* 0x04 */ s32 unk_04;
+    /* 0x08 */ u8 unk_08;
+    /* 0x0C */ f32 unk_0C;
+};
```

Comment:
```suggestion
struct struct_80B8E1A8 {
    /* 0x00 */ AnimationHeader* animationSeg;
    /* 0x04 */ f32 playbackSpeed;
    /* 0x08 */ u8 mode;
    /* 0x0C */ f32 transitionRate;
};
```
All the uses of `unk_04` looks like floats right? `0x3F800000 = 1.0f` Although I don't even see this used once in this file. Also, this struct seems familiar, is it common enough to add to z64animation.h? Open to thoughts

Reply:
Yeah. OoT seems to have one very similar.
I'll move the struct to that header then.